assembly/src/library/masl.rs

```diff
@@ -51,9 +53,10 @@ impl MaslLibrary {
     /// # Errors
     /// Returns an error if the provided `modules` vector is empty or contains more than
     /// [u16::MAX] elements.
-    fn new(
+    pub fn new(
         namespace: LibraryNamespace,
         version: Version,
+        locations: bool,
         modules: Vec<Module>,
     ) -> Result<Self, LibraryError> {
```

Comment:
Should we make this constructor private? Otherwise, it may be possible to create a `MaslLibrary` in a somewhat inconsistent state (i.e., `location = true`, but locations are not loaded).

Reply:
Makes sense; done in https://github.com/0xPolygonMiden/miden-vm/pull/903/commits/80babace5b0f2a202af6e3033f068e39c3e9d449

In fact, these APIs, if not public, aren't needed at all